### PR TITLE
#164069224 rate an article

### PR DESCRIPTION
**Pull request by wasibani-roy**
**What does this PR do**
This PR enables a user of the system to rate an article that they have reviewed as well as give an average rating to articles

**Description of Task to be completed?**

- Enable an authenticated user to rate an article

- Calculate an average rating for an article basing on ratings from users

- Display average rating for a given article

**How should this be manually tested?**

- Clone the repo and cd into Ah-backend-aquaman
- git checkout ft-user-rate-article-164069224
- pip install -r requirements.txt
- python manage.py makemigrations
- python manage.py migrate
- python manage.py runserver
- Register and then login into the system
- The below endpoint will be used to rate an article

Rate an article endpoint
POST http://127.0.0.1:8000/api/articles/articleslug/rate/

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**

- [#164069224](https://www.pivotaltracker.com/story/show/164069224)

**Screenshots**
**How to make your rating**
Screenshot 2019-03-13 at 15 19 05

**Response data**
Screenshot 2019-03-13 at 09 29 56

### CODE REVIEW

File: authors/apps/articles/tests/test_rating_article.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        self.add_article()
+        article = Article.objects.all().first()
+        self.register_and_login_new_user()
+        response = self.client.post(
+            reverse("articles:rating", kwargs={'slug': article.slug}),
+            data={"article": {"score": 6}},
+            format="json"
+        )
+        self.assertEqual(response.status_code, 400)
+        # self.assertIn("errors", response.json())
```
Comment (CryceTruly) [inline authors/apps/articles/tests/test_rating_article.py]: Thanks for the good work @wasibani-roy, however, line 60 does not look like a comment, it looks like commented code, you could remove the block for readability.

File: authors/apps/articles/tests/test_data.py [hunk 2]
```diff
@@ -6,3 +6,23 @@
         "body": "this is the body"
     }
 }
+updated_article = {
```
Comment (CryceTruly) [inline authors/apps/articles/tests/test_data.py]: Thanks for the good work @wasibani-roy, however, it looks like an updated article data is similar to valid article data, you could just one them to keep the code DRY.

File: authors/apps/articles/tests/test_rating_article.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+"""
+test module for rating an article
+"""
+from django.urls import reverse
+from authors.apps.articles.tests.base_class import ArticlesBaseTest
+from ..models import Article
+from .test_data import valid_article, updated_article
```
Comment (CryceTruly) [inline authors/apps/articles/tests/test_rating_article.py]: Hey @wasibani-roy, thanks for the good work, however on line 7 you import `valid_article` and `updated_article` but it seems you don't use them in this file.

**Review by CryceTruly — COMMENTED** [review]
Thanks for the good work Roy, you only need to remove files like admin.py, the tags from models and some code cleanup.

File: authors/apps/articles/tests/test_rating_article.py (hunk 1 above)
Comment (wasibani-roy) [reply authors/apps/articles/tests/test_rating_article.py]: Change made

File: authors/apps/articles/tests/test_data.py (hunk 2 above)
Comment (wasibani-roy) [reply authors/apps/articles/tests/test_data.py]: Done

File: authors/apps/articles/tests/test_rating_article.py (hunk 3 above)
Comment (wasibani-roy) [reply authors/apps/articles/tests/test_rating_article.py]: Worked on them

**Comment by Peace-Apple** [comment]
Good work Roy, from what I can see the structure is okay, am now testing it out locally on my machine and will submit my review in a bit.

File: authors/apps/articles/tests/test_data.py [hunk 4]
```diff
@@ -6,3 +6,5 @@
         "body": "this is the body"
     }
 }
+
+
```
Comment (Peace-Apple) [inline authors/apps/articles/tests/test_data.py]: Hello Roy could you keep just one blank at the end of this code as per the pep8 guidelines

**Review by Peace-Apple — DISMISSED** [review]
Good work Roy, thanks for working on the feedback

**Review by CryceTruly — DISMISSED** [review]
LGTM

**Review by BagzieGracious — DISMISSED** [review]

**Review by BagzieGracious — COMMENTED** [review]
Thank you for your goodwork @wasibani-roy, but can you work on your commits

**Review by CryceTruly — DISMISSED** [review]
LGTM

**Review by Peace-Apple — DISMISSED** [review]

**Review by BagzieGracious — APPROVED** [review]
LGTM

**Review by joelethan — APPROVED** [review]
lgtm

**Review by Peace-Apple — APPROVED** [review]

**Pull request merged**